### PR TITLE
*: move opt into clusterInfo

### PR DESCRIPTION
**Pull request by Connor1996**
This pr is for supporting specific for different namespace which introduces later.
Since namespace is transparent to scheduler,  but cluster does know, so move opt into clusterInfo, and then namespace's config is transparent to scheduler.

### CODE REVIEW

File: server/schedule/filters.go [hunk 1]
```diff
@@ -22,25 +22,25 @@ import (
 // Filter is an interface to filter source and target store.
 type Filter interface {
 	// Return true if the store should not be used as a source store.
-	FilterSource(store *core.StoreInfo) bool
+	FilterSource(store *core.StoreInfo, cluster Cluster) bool
```
Comment (disksing) [inline server/schedule/filters.go]: maybe better `cluster Cluster, store *core.StoreInfo`

File: server/cache_test.go [hunk 2]
```diff
@@ -531,7 +533,8 @@ func (s *testClusterInfoSuite) testRegionSplitAndMerge(c *C, cache *clusterInfo)
 }
 
 func (s *testClusterInfoSuite) TestUpdateStorePendingPeerCount(c *C) {
-	cluster := newClusterInfo(core.NewMockIDAllocator())
+	_, opt := newTestScheduleConfig()
+	cluster := newClusterInfo(core.NewMockIDAllocator(), opt, core.NewKV(core.NewMemoryKV()))
```
Comment (disksing) [inline server/cache_test.go]: I have noticed this use pattern appeared too many times in tests, consider to extract a function like 'newMockClusterInfo'.

File: server/schedulers/mockcluster.go [hunk 3]
```diff
@@ -26,14 +26,16 @@ import (
 
 type mockCluster struct {
 	*schedule.BasicCluster
-	id *core.MockIDAllocator
+	id  *core.MockIDAllocator
+	opt *MockSchedulerOptions
```
Comment (disksing) [inline server/schedulers/mockcluster.go]: Maybe make it an embedded field.

File: server/schedule/opts.go [hunk 4]
```diff
@@ -1,34 +0,0 @@
...
-// limitations under the License.
-
-package schedule
-
-import (
-	"time"
-)
-
-// Options for schedulers.
-type Options interface {
```
Comment (nolouch) [inline server/schedule/opts.go]: maybe we can embed this interface to Cluster, I think Filter/Selector use Options interface more clearly.

File: server/schedulers/mockcluster.go (hunk 3 above)
Comment (Connor1996) [reply server/schedulers/mockcluster.go]: next pr need opt

**Comment by disksing** [comment]
LGTM. Please resolve conflicts.

**Pull request merged**